File: app/src/main/java/social/entourage/android/involvement/GetInvolvedFragment.java
package social.entourage.android.involvement;


import android.content.ActivityNotFoundException;
import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;
import android.widget.Toast;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;
import social.entourage.android.BuildConfig;
import social.entourage.android.EntourageEvents;
import social.entourage.android.R;
import social.entourage.android.base.EntourageDialogFragment;

/**
 * Get Involved Menu Fragment
 */
public class GetInvolvedFragment extends EntourageDialogFragment {

    // ----------------------------------
    // CONSTANTS
    // ----------------------------------

    public static final String TAG = GetInvolvedFragment.class.getSimpleName();

    // ----------------------------------
    // ATTRIBUTES
    // ----------------------------------

    @BindView(R.id.get_involved_version)
    TextView versionTextView;

    // ----------------------------------
    // LIFECYCLE
    // ----------------------------------

    public GetInvolvedFragment() {
        // Required empty public constructor
    }

    public static GetInvolvedFragment newInstance() {
        return new GetInvolvedFragment();
    }


    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        super.onCreateView(inflater, container, savedInstanceState);
        // Inflate the layout for this fragment
        View view = inflater.inflate(R.layout.fragment_get_involved, container, false);

        ButterKnife.bind(this, view);

        return view;
    }

    @Override
    public void onViewCreated(@NonNull final View view, @Nullable final Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        populate();
    }

    private void populate() {
        versionTextView.setText(getString(R.string.about_version_format, if (BuildConfig.DEBUG) BuildConfig.VERSION_DEBUG_NAME else BuildConfig.VERSION_NAME));
    }

    // ----------------------------------
    // BUTTON HANDLING
    // ----------------------------------

    @OnClick(R.id.title_close_button)
    protected void onCloseButton() {
        dismiss();
    }

    @OnClick(R.id.get_involved_rate_us_layout)
    protected void onRateUsClicked() {
        EntourageEvents.logEvent(EntourageEvents.EVENT_ABOUT_RATING);

        Uri uri = Uri.parse(getString(R.string.market_url, this.getActivity().getPackageName()));
        Intent goToMarket = new Intent(Intent.ACTION_VIEW, uri);
        // To count with Play market backstack, After pressing back button,
        // to taken back to our application, we need to add following flags to intent.
        goToMarket.addFlags(Intent.FLAG_ACTIVITY_NO_HISTORY |
                Intent.FLAG_ACTIVITY_CLEAR_WHEN_TASK_RESET |
                Intent.FLAG_ACTIVITY_MULTIPLE_TASK);
        try {
            startActivity(goToMarket);
        } catch (ActivityNotFoundException e) {
            startActivity(new Intent(Intent.ACTION_VIEW,
                    Uri.parse(getString(R.string.playstore_url,this.getActivity().getPackageName()))));
        }
    }

    @OnClick(R.id.get_involved_facebook_layout)
    protected void onFacebookClicked() {
        EntourageEvents.logEvent(EntourageEvents.EVENT_ABOUT_FACEBOOK);

        Intent browserIntent = new Intent(Intent.ACTION_VIEW, Uri.parse(getString(R.string.facebook_url)));
        try {
            startActivity(browserIntent);
        } catch (ActivityNotFoundException ex) {
            Toast.makeText(getContext(), R.string.no_browser_error, Toast.LENGTH_SHORT).show();
        }
    }

    @OnClick(R.id.get_involved_instagram_layout)
    protected void onInstaClicked() {
        Intent browserIntent = new Intent(Intent.ACTION_VIEW, Uri.parse(getString(R.string.instagram_url)));
        try {
            startActivity(browserIntent);
        } catch (ActivityNotFoundException ex) {
            Toast.makeText(getContext(), R.string.no_browser_error, Toast.LENGTH_SHORT).show();
        }
    }

    @OnClick(R.id.get_involved_twitter_layout)
    protected void onTwitterClicked() {
        Intent browserIntent = new Intent(Intent.ACTION_VIEW, Uri.parse(getString(R.string.twitter_url)));
        try {
            startActivity(browserIntent);
        } catch (ActivityNotFoundException ex) {
            Toast.makeText(getContext(), R.string.no_browser_error, Toast.LENGTH_SHORT).show();
        }
    }

}

File: app/src/main/java/social/entourage/android/involvement/GetInvolvedFragment.kt
package social.entourage.android.involvement

import android.content.ActivityNotFoundException
import android.content.Intent
import android.net.Uri
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Toast
import kotlinx.android.synthetic.main.fragment_get_involved.*
import kotlinx.android.synthetic.main.layout_view_title.*
import social.entourage.android.BuildConfig
import social.entourage.android.tools.log.AnalyticsEvents
import social.entourage.android.R
import social.entourage.android.base.BaseDialogFragment

/**
 * Get Involved Menu Fragment
 */
class GetInvolvedFragment  : BaseDialogFragment() {
    // ----------------------------------
    // LIFECYCLE
    // ----------------------------------
    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        super.onCreateView(inflater, container, savedInstanceState)
        // Inflate the layout for this fragment
        return inflater.inflate(R.layout.fragment_get_involved, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        populate()
        title_close_button?.setOnClickListener { onCloseButton() }
        get_involved_rate_us_layout?.setOnClickListener { onRateUsClicked() }
        get_involved_facebook_layout?.setOnClickListener { onFacebookClicked() }
        get_involved_instagram_layout?.setOnClickListener { onInstaClicked() }
        get_involved_twitter_layout?.setOnClickListener { onTwitterClicked() }
    }

    private fun populate() {
        get_involved_version?.text = getString(R.string.about_version_format, BuildConfig.VERSION_FULL_NAME)
    }

    // ----------------------------------
    // BUTTON HANDLING
    // ----------------------------------
    fun onCloseButton() {
        dismiss()
    }

    fun onRateUsClicked() {
        activity?.let {
            AnalyticsEvents.logEvent(AnalyticsEvents.EVENT_ABOUT_RATING)
            val goToMarket = Intent(Intent.ACTION_VIEW,
                    Uri.parse(getString(R.string.market_url, it.packageName)))
            // To count with Play market backstack, After pressing back button,
            // to taken back to our application, we need to add following flags to intent.
            goToMarket.addFlags(Intent.FLAG_ACTIVITY_NO_HISTORY or
                    Intent.FLAG_ACTIVITY_CLEAR_WHEN_TASK_RESET or
                    Intent.FLAG_ACTIVITY_MULTIPLE_TASK)
            try {
                startActivity(goToMarket)
            } catch (e: ActivityNotFoundException) {
                startActivity(Intent(Intent.ACTION_VIEW,
                        Uri.parse(getString(R.string.playstore_url, it.packageName))))
            }
        }
    }

    fun onFacebookClicked() {
        AnalyticsEvents.logEvent(AnalyticsEvents.EVENT_ABOUT_FACEBOOK)
        val browserIntent = Intent(Intent.ACTION_VIEW, Uri.parse(getString(R.string.facebook_url)))
        try {
            startActivity(browserIntent)
        } catch (ex: ActivityNotFoundException) {
            Toast.makeText(context, R.string.no_browser_error, Toast.LENGTH_SHORT).show()
        }
    }

    fun onInstaClicked() {
        val browserIntent = Intent(Intent.ACTION_VIEW, Uri.parse(getString(R.string.instagram_url)))
        try {
            startActivity(browserIntent)
        } catch (ex: ActivityNotFoundException) {
            Toast.makeText(context, R.string.no_browser_error, Toast.LENGTH_SHORT).show()
        }
    }

    fun onTwitterClicked() {
        val browserIntent = Intent(Intent.ACTION_VIEW, Uri.parse(getString(R.string.twitter_url)))
        try {
            startActivity(browserIntent)
        } catch (ex: ActivityNotFoundException) {
            Toast.makeText(context, R.string.no_browser_error, Toast.LENGTH_SHORT).show()
        }
    }

    companion object {
        // ----------------------------------
        // CONSTANTS
        // ----------------------------------
        val TAG = GetInvolvedFragment::class.java.simpleName
        fun newInstance(): GetInvolvedFragment {
            return GetInvolvedFragment()
        }
    }
}